src/main/java/co/rsk/federate/signing/hsm/message/ReleaseCreationInformation.java

```diff
@@ -48,6 +58,29 @@ public ReleaseCreationInformation(
         this.pegoutCreationRskTxHash = pegoutCreationRskTxHash;
         this.pegoutBtcTx = pegoutBtcTx;
         this.pegoutConfirmationRskTxHash = pegoutConfirmationRskTxHash;
+
+        this.decodeUtxoOutpointValues(transactionReceipt);
+    }
+
+    private void decodeUtxoOutpointValues(TransactionReceipt transactionReceipt) {
```

Comment:
What about making this method return `utxoOutpointValues` so we are not setting instance variables as side effects. Something like:

```suggestion
        this.utxoOupointValues = decodeUtxoOutpointValues(transactionReceipt);
    }

    private List<Coin> decodeUtxoOutpointValues(TransactionReceipt transactionReceipt) {
```

Reply:
Nice idea.